obspy/io/gcf/core.py

```diff
@@ -372,19 +452,27 @@ def _read_gcf(filename, headonly=False, network='', station='',
 
     # read file
     obj = _GcfFile()
-    b_filename = filename.encode('utf-8')
-    ret = gcf_io.read_gcf(b_filename, obj, mode)
-    if ret == -1:
-        if os.path.isfile(filename):
-            raise IOError("cannot open file %s in read mode (missing "
-                          "read permissions for user?)" % (filename))
-        else:
-            raise IOError("%s is not a file" % (filename))
-    elif ret == 1:
-        raise IOError("file %s is not a GCF data file" % (filename))
-    elif ret:
-        raise IOError("failed to read file %s (unknown error code %d "
-                      "returned)" % (filename, ret))
+    # b_filename = filename.encode('utf-8')
+    # ret = gcf_io.read_gcf(fid, obj, mode)
+    ret = _fill_gcf_from_file(file, obj, mode)
+
+    if isinstance(file, IOBase):
+        if ret:
+            raise IOError("failed to read in-memory GCF data (error code %d)"
```

Comment:
if it's `IOBase`, isn't it "in-memory" misleading? Looks more like a file handle, potentially or in general?